File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/pvp/AlignmentSubAreaUpdateMessage.scala





















// Generated on 05/03/2013 10:34:59
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait AlignmentSubAreaUpdateMessage extends MessageDeserializer {
  val messageId = 6057

  protected trait Message extends BaseMessage with Serializable {
    val messageId = AlignmentSubAreaUpdateMessage.messageId

    val subAreaId: Short
    val side: Byte
    val quiet: Boolean

    def serialize(buf: Buffer) {
      buf.writeShort(subAreaId)
      buf.writeByte(side)
      buf.writeBoolean(quiet)
    }

  }

  protected def create(subAreaId: Short, side: Byte, quiet: Boolean): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val subAreaId = buf.readShort()
    val side = buf.readByte()
    val quiet = buf.readBoolean()

    create(subAreaId, side, quiet)
  }

}

object AlignmentSubAreaUpdateMessage extends AlignmentSubAreaUpdateMessage {
  case class MessageImpl(subAreaId: Short, side: Byte, quiet: Boolean) extends Message
  type Target = MessageImpl

  override def create(subAreaId: Short, side: Byte, quiet: Boolean) = MessageImpl(subAreaId, side, quiet)

  def apply(subAreaId: Short, side: Byte, quiet: Boolean) = create(subAreaId, side, quiet)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/AlignmentSubAreaUpdateMessage.java


// Generated on 05/08/2013 19:37:58
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class AlignmentSubAreaUpdateMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6057;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short subAreaId;
    public byte side;
    public boolean quiet;
    
    public AlignmentSubAreaUpdateMessage() { }
    
    public AlignmentSubAreaUpdateMessage(short subAreaId, byte side, boolean quiet) {
        this.subAreaId = subAreaId;
        this.side = side;
        this.quiet = quiet;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(subAreaId);
        buf.writeByte(side);
        buf.writeBoolean(quiet);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        subAreaId = buf.readShort();
        if (subAreaId < 0)
            throw new RuntimeException("Forbidden value on subAreaId = " + subAreaId + ", it doesn't respect the following condition : subAreaId < 0");
        side = buf.readByte();
        quiet = buf.readBoolean();
    }
    
}
